### ISSUE TITLE
Make it easy for users to know how to treat `inline-config` file and its lock file

### ISSUE THREAD

**Issue by lread** [clj-kondo/clj-kondo#2606]
- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**version**
```
clj-kondo v2025.07.28
```

**platform**
native on linux

**problem**
When using clj-kondo inline config, after a lint run, I see some unexpected files after the run is complete.

**repro**
From a new empty dir:
```shell
mkdir .clj-kondo
mkdir -p src/foo
cat >> src/foo/bar.clj <<EOF
(ns foo.bar)

(defn- #_:clj-kondo/ignore booya [])
EOF
```

Run lint:
```shell
clj-kondo --lint src
```
Outputs then exits with 0:
```
linting took 11ms, errors: 0, warnings: 0
```

List files in `.clj-kondo`
```shell
ls -la .clj-kondo
```

**actual behaviour**
I see:
```
drwxrwxr-x 4 lee lee 4096 Aug 14 10:01 .
drwxrwxr-x 4 lee lee 4096 Aug 14 10:12 ..
drwxrwxr-x 3 lee lee 4096 Aug 14 10:01 .cache
drwxrwxr-x 3 lee lee 4096 Aug 14 10:01 inline-configs
-rw-rw-r-- 1 lee lee    0 Aug 14 10:01 .lock
```

**expected behavior**
I expected to only see the `.cache` directory.

Is the generated `inline-config` normal for this scenario?

The `.lock` file seems like it should not be there.
 
**next up**
Happy to help if this is an issue.

**Comment by borkdude** [comment]
The `inline-config` directory is normal, it is where it stores extracted configurations from namespaces.
The `.lock` file is created during the reading and writing of the inline configs. So this is all expected.

**Comment by borkdude** [comment]
Perhaps it would have been more logical to create the `.lock` file inside of `inline-configs` though.

**Comment by lread** [comment]
Ah! Thank you for the reply!

I could be wrong, but these feel like cache files because they will get regenerated if they do not exist.
Could they somehow be plunked under `.clj-kondo/.cache`?

A benefit of placing them under `.cache` is that many folks already `.gitignore` the the `.cache`.

**Comment by borkdude** [comment]
No, the inline-config falls in the category of imported configurations, not cached analysis data.

**Comment by lread** [comment]
But if we squint (maybe I've been doing much squint cljs!) could we see these as cache files in the uncategorized broad sense? 

If that does not jibe, I can update advice in docs around what folks might want to check into version control and what they might want to git ignore. 

Like `.cache`, I think `.lock` and  `inline-configs` would generally be git ignored?

**Comment by borkdude** [comment]
`inline-configs` has the same status as `imports`. They are "imported" when you lint dependencies. So they should be treated the same way. Files starting with `.` can generally be git-ignored.

**Comment by lread** [comment]
Okay, thanks, I'm going to rename this to a documentation issue.

**Comment by borkdude** [comment]
I think still moving the `.lock` file is a good idea though. we could even move it inside `.cache` so people just can ignore one directory

**Comment by lread** [comment]
Can do a PR for that if you like. Is the `.lock` file generic or should it be renamed?

**Comment by borkdude** [comment]
We can move it inside `.cache` but `.cache` already has a `.lock` file for protecting writes to the `.cache` directory.  So maybe better rename it to `.cache/.inline-config-lock`

**Comment by lread** [comment]
Hmmm... user can specify `--cache false`.

Curiosity: In general, when I see a `.lock` file after completion, I assume a crash. Can we delete the `.lock` file? Or does it need to persist after a clj-kondo run is complete?

**Comment by borkdude** [comment]
>  Can we delete the .lock file? 

No since multiple clj-kondo processes might be using it.

**Comment by lread** [comment]
Maybe some options would help.

## Option 1: do nothing
Cons:
- affected users will be confused what to do with `.clj-kondo/.lock` and `.clj-kondo/inline-config`

## Option 2: document only
Cons:
- affected users will have to add git ignore for `.lock`, which seems not too arduous!

## Option 3: move `.lock` under `.cache` (and rename it)
Cons:
- users can specify `--cache false` what then?

Pros: 
- users already typically git ignore `.cache` 

## Option 4: move `.lock` under `inline-config` dir
Cons:
- Since users will be checking in `inline-config`, they will also be checking in the `.lock`, is that a very bad thing?

## Option 5: rename `.lock` to `.inline-config-lock` and document
Cons:
- less generic to git ignore, but not by much: a generic `.*lock` or an explicit `.inline-config-lock`

Pros:
- self-documents what the lock is for

## Proposal
Maybe 2 or 5? Whatcha think?

**Comment by borkdude** [comment]
Let's name the file `.config-lock` and put it into `.clj-kondo/.cache`. Create the directory when it doesn't exist. When `--cache false` is active, it doesn't matter if we still create that directory I think?

**Comment by lread** [comment]
To confirm: always plunk lock under `.clj-kondo/.cache/` regardless of `--cache` and `--cache-dir`.

That works. It's a constant spot, so `inline-config` will be protected when multiple processes are running.

**Comment by borkdude** [comment]
Yes, that's the idea...
